calculator.php

```diff
@@ -125,45 +125,60 @@ function finish_app()
     }
 }
 
-function show_info_block()
-{
-    info('Calculator commands: ' . (implode(' ; ', AVAILABLE_COMMANDS)) . ' ;');
-    info('History commands:' . PHP_EOL . 'Full - to see full history.' . PHP_EOL . 'Format of date "01-01-1990" - to show history of current date.');
-}
-
 function show_history($historyRepository)
 {
     $history = $historyRepository->all();
 
     if (empty($history)) {
         return info('You have no history');
     }
-    $historyGroups = array_group($history, 'date');
 
-    $historyCommands = array_merge(['full', 'help'], array_keys($historyGroups));
-    $showDateHistory = readline('Enter date of history (DD-MM-YYYY), "Full" or "Help"  : ');
+    do {
+        $historyGroups = array_group($history, 'date');
 
-    $isDate = date_create_from_format('j-m-Y', $showDateHistory);
+        $historyCommands = array_merge(['full', 'help', 'back'], array_keys($historyGroups));
+        $showDateHistory = readline('Enter date of history (DD-MM-YYYY), "full" or "help"  : ');
+        $showDateHistory = strtolower($showDateHistory);
```

Comment:
let's implement "ask" console helper which will make read line and make it lower case